BlueprintUI/Sources/Environment/Environment.swift

```diff
@@ -34,33 +34,103 @@ import UIKit
 ///             }
 ///         }
 ///     }
-public struct Environment {
+public struct Environment: Equatable {
+
+    typealias OnDidRead = (StorageKey) -> Void
+
+    enum LayoutPass: Equatable {
+        case measurement
+        case layout
+    }
 
     /// A default "empty" environment, with no values overridden.
     /// Each key will return its default value.
     public static let empty = Environment()
 
-    private var values: [ObjectIdentifier: Any] = [:]
+    /// When enabled, notify any subscribers when a value is read
+    /// from the environment
+    var readNotificationsEnabled: Bool = true
```

Comment:
When would you want to disable this? Could we just rely on the presence of subscriptions instead?

Reply:
We turn this off when adapting the environment so erroneous reads aren't fired; which is coming in a later PR: https://github.com/square/Blueprint/pull/367/files#diff-07c71c5dcdc5da4c2ddf51b560695e562af396d4afafe3908b35656a0907459aR568

Reply:
Update: Since this a relatively small optimization; we just removed this in a later PR.